### PR TITLE
Check isStop at start of loops in ascanrules

### PR DESCRIPTION
**Pull request by FiveOFive**
Fixes [zaproxy/zaproxy#6647](https://github.com/zaproxy/zaproxy/issues/6647)

The original issue was only for the CodeInjectionScanRule. This change also updates other rules that had the same problem.

Doesn't look like any of the rules currently unit test responding to stops and I did not see a straightforward way to add tests. Please let me know if anyone with more experience on the code base has thoughts for testing.

### CODE REVIEW

**Comment by FiveOFive** [comment]
Thanks. Think I got the sign-off added right by forcing a new commit.

**Comment by thc202** [comment]
Note that you don't need to add the link yourself: https://docs.github.com/en/github/managing-your-work-on-github/managing-your-work-with-issues-and-pull-requests/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

File: addOns/ascanrules/CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 ### Fixed
 - Correct Context check in SQL Injection scan rule.
 - "Source Code Disclosure - /WEB-INF folder" is no longer skipped on Java 9+ (Issue 4038).
+- isStop() not checked on IOException leads to max rule duration not enforced (Issue 6647).
```
Comment (thc202) [inline addOns/ascanrules/CHANGELOG.md:13]: I'd rephrase it to not refer to the Java method, user does not necessarily know about it.

File: addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CodeInjectionScanRule.java [hunk 2]
```diff
@@ -169,6 +169,13 @@ public void scan(HttpMessage msg, String paramName, String value) {
      */
     private boolean testPhpInjection(String paramName) {
         for (String phpPayload : PHP_PAYLOADS) {
+            // Check if the scan has been stopped
+            // if yes dispose resources and exit
```
Comment (thc202) [inline addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CodeInjectionScanRule.java:173]: Good opportunity to remove these unnecessary comments (same for the one in the if).

**Comment by thc202** [comment]
> Please let me know if anyone with more experience on the code base has thoughts for testing.

I guess change the server to cause an I/O exception while stopping the rule and then verify that it does not do any more requests.

**Comment by thc202** [comment]
@FiveOFive are you available to address the comments or should we go ahead?

File: addOns/ascanrules/CHANGELOG.md (hunk 1 above)
Comment (FiveOFive) [reply addOns/ascanrules/CHANGELOG.md]: Updated

File: addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CodeInjectionScanRule.java (hunk 2 above)
Comment (FiveOFive) [reply addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CodeInjectionScanRule.java]: Removed the comments

**Comment by FiveOFive** [comment]
> @FiveOFive are you available to address the comments or should we go ahead?

Yes! Got a bit busy over the weekend, but back now. Just pushed an update to address comments.

**Review by kingthorin — APPROVED** [review]
Thanks!

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!
